spyder/plugins/maininterpreter.py

```diff
@@ -153,6 +158,28 @@ def python_executable_switched(self, custom):
             if custom:
                 self.warn_python_compatibility(cust_pyexec)
 
+    def is_python_interpreter(self, pyexec):
+        if not osp.isfile(pyexec):
+            return False
+        args = ["-c", "print('valid_python_interpreter')"]

```

Comment:
Problem is this also works for `ipython` (although we could also filter for 2.7.x and 3.{4,5,6})

Besides, most Linux programs have a `--version` option and some could coincide with the Python versions I mentioned above :-)